### PR TITLE
Build fix for RIOT target after the modifications of the build system…

### PR DESCRIPTION
**Pull request by robertsipka**
… (commit ddab1d8).

JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

File: targets/riot-stm32f4/Makefile.riot [hunk 1]
```diff
@@ -29,24 +27,24 @@ EXT_CFLAGS += -Wno-error=format=
...
-	cmake -B$(INTERM) -H./ \
-	 -DEXTERNAL_PORT_DIR=UNDEFINED \
+	cmake -B$(BUILD_DIR) -H./ \
 	 -DENABLE_LTO=OFF \
-	 -DENABLE_VALGRIND=OFF \
-	 -DCMAKE_TOOLCHAIN_FILE=build/configs/toolchain_external.cmake \
+	 -DFEATURE_VALGRIND=OFF \
+	 -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain_external.cmake \
+	 -DJERRY_LIBC=OFF\
+	 -DJERRY_CMDLINE=OFF\
```
Comment (polaroi8d) [inline targets/riot-stm32f4/Makefile.riot]: There is need a whitespace, or not?

Comment (zherczeg) [inline targets/riot-stm32f4/Makefile.riot]: Probably optional, but I agree that it looks better with white space.

File: targets/riot-stm32f4/Makefile.riot [hunk 2]
```diff
@@ -29,24 +27,24 @@ EXT_CFLAGS += -Wno-error=format=
...
 	 -DCOMPILER_DEFAULT_LIBC=ON \
+	 -DENABLE_ALL_IN_ONE=OFF\
```
Comment (zherczeg) [inline targets/riot-stm32f4/Makefile.riot]: Ditto.

**Comment by zherczeg** [comment]
LGTM after the spaces added.

**Comment by robertsipka** [comment]
Thanks, I've updated the patch.

File: targets/riot-stm32f4/Makefile.riot [hunk 3]
```diff
@@ -29,24 +27,24 @@ EXT_CFLAGS += -Wno-error=format=
...
 	 -DEXTERNAL_CMAKE_C_COMPILER_ID=GNU \
 	 -DEXTERNAL_COMPILE_FLAGS="$(EXT_CFLAGS)" \
-	 -DEXTERNAL_MEM_HEAP_SIZE_KB=$(JERRYHEAP)
+	 -DMEM_HEAP_SIZE_KB=$(JERRYHEAP)
 
-	make -C $(INTERM) $(TYPE).external
-	cp `cat $(INTERM)/$(TYPE).external/list` $(OUTPUT)/.
-	cp $(OUTPUT)/lib$(TYPE).jerry-core.a $(COPYTARGET)/libjerrycore.a
+	make -C$(BUILD_DIR) jerry-core
+	cp $(BUILD_DIR)/lib/libjerry-core.a $(COPYTARGET)/libjerrycore.a
```
Comment (polaroi8d) [inline targets/riot-stm32f4/Makefile.riot]: `$(BUILD_DIR) = build/riotstm32f4/` So after the cp command ..//lib/libjerry-core.a ...  is it okey?

Comment (robertsipka) [inline targets/riot-stm32f4/Makefile.riot]: It is, but really ugly, I fixed it, thanks.

**Comment by LaszloLango** [comment]
LGTM

**Comment by robertsipka** [comment]
I've rebased it with master.

**Pull request merged**